controllers/nnf_workflow_controller_helpers.go

```diff
@@ -1325,3 +1273,155 @@ func (r *NnfWorkflowReconciler) createPinnedContainerProfileIfNecessary(ctx cont
 
 	return nil
 }
+
+func addPinnedContainerProfileLabel(object metav1.Object, nnfContainerProfile *nnfv1alpha1.NnfContainerProfile) {
+	labels := object.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+
+	labels[nnfv1alpha1.PinnedContainerProfileLabelName] = nnfContainerProfile.GetName()
+	labels[nnfv1alpha1.PinnedContainerProfileLabelNameSpace] = nnfContainerProfile.GetNamespace()
+	object.SetLabels(labels)
+}
+
+// Create a list of volumes to be mounted inside of the containers based on the DW_JOB/DW_PERSISTENT arguments
+func (r *NnfWorkflowReconciler) getContainerVolumes(ctx context.Context, workflow *dwsv1alpha1.Workflow, dwArgs map[string]string, nnfNodeName string) ([]nnfContainerVolume, error) {
```

Comment:
@matthew-richerson Aren't all the Rabbit mounts located at the same path? I thought we'd be able to get the NNF Access that describes the mount once, and use that mountpoint for all container volumes.

Reply:
Yes, you could just look at the NnfAccess. LLNL wanted the index to compute node name translation for gfs2 file systems, and to do that we'll need to look at the individual ClientMount resources. So I think this code will be useful in the future.

Reply:
I'll give this a go and only use the NnfAccess. Even though I talked to @matthew-richerson about this, I think I misunderstood what he was saying in the first place.

Reply:
> LLNL wanted the index to compute node name translation for gfs2 file systems, and to do that we'll need to look at the individual ClientMount resources

I think they wanted links, which would be an enhancement to the NNF client mount controller to create when the mount occurs. Maybe I'm missing something, but I don't see container logic ever needing to get the client mounts.

Reply:
I've updated this to use NNFAccess rather than ClientMount.